Adyen/HttpClient/HttpClientWrapper.cs

```diff
@@ -0,0 +1,140 @@
+﻿#region License
+
+// /*
+//  *                       ######
+//  *                       ######
+//  * ############    ####( ######  #####. ######  ############   ############
+//  * #############  #####( ######  #####. ######  #############  #############
+//  *        ######  #####( ######  #####. ######  #####  ######  #####  ######
+//  * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
+//  * ###### ######  #####( ######  #####. ######  #####          #####  ######
+//  * #############  #############  #############  #############  #####  ######
+//  *  ############   ############  #############   ############  #####  ######
+//  *                                      ######
+//  *                               #############
+//  *                               ############
+//  *
+//  * Adyen Dotnet API Library
+//  *
+//  * Copyright (c) 2020 Adyen B.V.
+//  * This file is open source and available under the MIT license.
+//  * See the LICENSE file for more info.
+//  */
+
+#endregion
+
+using Adyen.Constants;
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using System.Web;
+using Adyen.HttpClient.Interfaces;
+using Adyen.Model;
+
+namespace Adyen.HttpClient
+{
+    public class HttpClientWrapper : IClient
+    {
+        private readonly Config _config;
+
+        private readonly Encoding _encoding = Encoding.UTF8;
+        private readonly System.Net.Http.HttpClient _httpClient;
+
+        public HttpClientWrapper(Config config, System.Net.Http.HttpClient httpClient)
+        {
+            //Set security protocol. Only TLS1.2
+            ServicePointManager.SecurityProtocol = SecurityProtocolType.Tls12;
+
+            _config = config;
+            _httpClient = httpClient;
+        }
+
+        public string Request(string endpoint, string json, bool isApiKeyRequired, RequestOptions requestOptions = null)
+        {
+            return RequestAsync(endpoint, json, isApiKeyRequired, requestOptions).GetAwaiter().GetResult();
+        }
+
+        public async Task<string> RequestAsync(string endpoint, string json, bool isApiKeyRequired, RequestOptions requestOptions = null)
+        {
+            using (var request = GetHttpRequestMessage(endpoint, isApiKeyRequired, json, requestOptions))
+            using (var httpResponseMessage = await _httpClient.SendAsync(request))
+            {
+                var responseText = await httpResponseMessage.Content.ReadAsStringAsync();
+                httpResponseMessage.EnsureSuccessStatusCode();
+                return responseText;
```

Comment:
There is no exception handling here. Is there any reason ?

Reply:
`EnsureSuccessStatusCode` throwing an exception that could be handled by the user if the status code is not succesful. What kind of handling are you expecting here?

Reply:
Indeed. We changed the exception handling on the [old HttpURLConnectionClient](https://github.com/Adyen/adyen-dotnet-api-library/blob/develop/Adyen/HttpClient/HttpURLConnectionClient.cs#L118). I was thinking if the `HandleWebException` is outsourced in another class if could we use the same here.